GVRf/Sample/gvrbulletsample/src/org/gearvrf/bulletsample/BulletSampleViewManager.java

```diff
@@ -0,0 +1,219 @@
+         */
+        mBullet.createPhysicsWorld(new Vector3(-480.0f, -480.0f, -480.0f),
+                new Vector3(480.0f, 480.0f, 480.0f), 1024, new Vector3(0.0f,
+                        -9.8f, 0.0f));
+
+        /*
+         * Create the ground. A simple textured quad. In bullet it will be a
+         * plane shape with 0 mass
+         */
+        GVRSceneObject groundScene = quadWithTexture(100.0f, 100.0f,
+                "floor.jpg");
+        groundScene.getTransform().setRotationByAxis(-90.0f, 1.0f, 0.0f, 0.0f);
+        groundScene.getTransform().setPosition(0.0f, 0.0f, 0.0f);
+        scene.addSceneObject(groundScene);
+
+        StaticPlaneShape floorShape = new StaticPlaneShape(new Vector3(0.0f,
+                1.0f, 0.0f), 0.0f);
+        Geometry floorGeometry = mBullet.createGeometry(floorShape, 0.0f,
+                new Vector3(0.0f, 0.0f, 0.0f));
+        MotionState floorState = new MotionState();
+        mBullet.createAndAddRigidBody(floorGeometry, floorState);
+
+        /*
+         * Create Some cubes in Bullet world and hit it with a sphere
+         */
+        addCube(scene, 0.0f, 1.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 0.0f, 1.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 0.0f, 1.0f, -11.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 1.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 1.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 1.0f, -11.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 1.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 1.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 1.0f, -11.0f, CUBE_MASS);
+
+        addCube(scene, 0.0f, 2.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 0.0f, 2.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 0.0f, 2.0f, -11.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 2.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 2.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 2.0f, -11.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 2.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 2.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 2.0f, -11.0f, CUBE_MASS);
+
+        addCube(scene, 0.0f, 3.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 0.0f, 3.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 0.0f, 3.0f, -11.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 3.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 3.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 1.0f, 3.0f, -11.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 3.0f, -9.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 3.0f, -10.0f, CUBE_MASS);
+        addCube(scene, 2.0f, 3.0f, -11.0f, CUBE_MASS);
+
+        /*
+         * Throw a sphere from top
+         */
+        addSphere(scene, 1.0f, 1.5f, 100.0f, -10.0f, 20.0f);
+    }
+
+    @Override
+    public void onStep() {
+        mBullet.doSimulation(1.0f / 60.0f, 10);
+        for (RigidBody body : rigidBodiesSceneMap.keySet()) {
+            if (body.geometry.shape.getType() == ShapeType.SPHERE_SHAPE_PROXYTYPE) {
```

Comment:
Minor code redundancy, the following change would look better for me: 
            if (body.geometry.shape.getType() == ShapeType.SPHERE_SHAPE_PROXYTYPE
                    || body.geometry.shape.getType() == ShapeType.BOX_SHAPE_PROXYTYPE) {

	// the second code block can be deleted.